docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
After thinking about it over night, I'd like to summarize what I think are a couple of possibilities for an alternate API:

1) Make the relative small change proposed above to combine exchange() and agree() on the private key. Typical usage would be:

```python
    shared_key = private_key.exchange(peer_public_key, ECDH(KDF(a, b, c)))
```

The KDF argument to ECDH() could default to ECDHnoKDF() so that this could be left off when not needed. The ephemeral ECDH case with no KDF would look something like:

```python
    private_key = ec.generate_private_key(curve, backend)
    public_key = private_key.public_key()   # Exchange this with peer
    shared_key = private_key.exchange(peer_public_key, ECDH())
```

2) Create separate ECDH and ECDHE (ephemeral ECDH) classes with a common API but different constructors:

```python
class ECDH:
    def __init__(self, private_key):
        """Begin an ECDH key exchange"""

    def get_public(self):
        """Return the public key to send to the peer"""

    def get_shared(self, peer_public_key, kdf):
        """Return the shared key given the peer's public key"""

class ECDHE(ECDH):
    def __init__(self, curve, backend):
        """Begin an ephemeral ECDH key exchange"""
```

The kdf argument to get_shared() could again default to ECDHnoKDF(). The ECDHE() variant would simply construct a random private key to pass to the parent's constructor. We may need to pass in a backend object to allow that.

In this case, ephemeral ECDHE with no KDF would look something like:

```python
    ecdh = ECDHE(curve, backend)
    public_key = ecdh.get_public()    # Exchange this with peer
    shared_key = ecdh.get_shared(peer_public_key)
```

The non-ephemeral one would look the same, except that a private_key would be passed to ECDH() instead of the curve & backend to ECDHE().